op-conductor/conductor/service_test.go

```diff
@@ -911,3 +917,192 @@ func (s *OpConductorTestSuite) TestSupervisorConnectionDown() {
+			CompressionMode: websocket.CompressionDisabled,
+		})
+		if err != nil {
+			s.T().Logf("Failed to accept WebSocket connection: %v", err)
+			return
+		}
+		defer conn.Close(websocket.StatusNormalClosure, "test complete")
+
+		// Signal that connection is established (only once)
+		serverConnectedOnce.Do(func() {
+			close(serverConnected)
+		})
+
+		// Send test messages immediately and signal completion
+		messages := []string{"Hello", "World", "Test"}
+		for _, msg := range messages {
+			err := conn.Write(testCtx, websocket.MessageText, []byte(msg))
+			if err != nil {
+				s.T().Logf("Failed to write message: %v", err)
+				return // Connection closed
+			}
+		}
+
+		// Signal messages sent (only once)
+		messagesSentOnce.Do(func() {
+			close(messagesSent)
+		})
+
+		// Keep connection alive by reading until context is cancelled
+		for {
+			select {
+			case <-testCtx.Done():
+				return
+			default:
+				// Read with timeout to avoid blocking indefinitely
+				readCtx, cancel := context.WithTimeout(testCtx, 100*time.Millisecond)
+				_, _, err := conn.Read(readCtx)
+				cancel()
+
+				if err != nil {
+					// Expected on timeout or connection close
+					if errors.Is(err, context.DeadlineExceeded) {
+						continue // Timeout is expected, continue loop
+					}
+					return // Other errors mean connection is closed
+				}
+			}
+		}
+	}))
+	defer rollupBoostServer.Close()
+
+	// Convert HTTP URL to WebSocket URL for rollup boost
+	rollupBoostWsURL := strings.Replace(rollupBoostServer.URL, "http", "ws", 1)
+
+	// Update the config to include the WebSocket URL and server port
+	s.cfg.RollupBoostWsURL = rollupBoostWsURL
+	s.cfg.WebsocketServerPort = port
+
+	// Create a new conductor with the updated config
+	conductor, err := NewOpConductor(s.ctx, &s.cfg, s.log, s.metrics, s.version, s.ctrl, s.cons, s.hmon)
+	s.NoError(err)
+
+	// Set up mock expectation for Leader() calls - the flashblocks handler checks leadership
+	// before forwarding messages, so we need to mock this to return true
+	s.cons.EXPECT().Leader().Return(true)
+
+	// Start the conductor, which should initialize and start the flashblocks handler
+	s.hmon.EXPECT().Start(mock.Anything).Return(nil)
+	err = conductor.Start(s.ctx)
+	s.NoError(err)
+
+	// Wait for conductor to be ready using its internal state
+	s.NotNil(conductor.flashblocksHandler, "flashblocks handler should be initialized")
+
+	// Wait for rollup boost server connection (event-driven, not time-based)
+	select {
+	case <-serverConnected:
+		// Connection established
+	case <-time.After(5 * time.Second):
+		s.Fail("Timeout waiting for rollup boost server connection")
+	}
+
+	// Wait for messages to be sent (event-driven)
+	select {
+	case <-messagesSent:
+		// Messages sent
+	case <-time.After(2 * time.Second):
+		s.Fail("Timeout waiting for messages to be sent")
+	}
+
+	// Connect to the WebSocket server using event-driven approach
+	wsURL := fmt.Sprintf("ws://localhost:%d/ws", s.cfg.WebsocketServerPort)
```

Comment:
IIUC by this line of code, op-conductor is started with no clients and it would have already listened to the 3 messages from rollupBoostServer, and also broadcasted them instantly to the clients it was aware of i.e. zero clients.

Meaning those messages would have gone by now.

So from the next line, when we create a client, how does that a client receive the messages from op-conductor as those messages are already gone?